Bug 1142147 Content Services pages

sgarrity opened the pull request:
Still in progress, but would like initial review on the implementation of the form at /contentservices/start/

It is based on the form at /about/partnerships/

CC: @pmclanahan

pmclanahan commented on bedrock/mozorg/forms.py:
```diff
@@ -506,6 +509,280 @@ def __init__(self, *args, **kwargs):
 
 
+class ContentServicesForm(forms.Form):
+    states = (
```
I don't see where this list of states is used.

sgarrity replied:
This was leftover cruft. Removed! Thanks.

pmclanahan commented:
Sorry about taking forever on this @sgarrity. Things are looking good though. Do you still need me to help with the blank initial option for the state field?

sgarrity commented:
> Sorry about taking forever on this @sgarrity. Things are looking good though. Do you still need me to help with the blank initial option for the state field?

No worries. We had been waiting on content in the meantime anyhow. Yes, if you can help with the blank initial option for the state field, that would be helpful.

sgarrity commented:
@pmclanahan Also, when you select United States as the country, the Province field hides and the State field appears. This is working for me locally, but not on demo3. Any idea why that might be?

pmclanahan commented:
@sgarrity it's because the output file name for the form JS wasn't changed and is being overwritten by the other "partnerships-bundle.js" file.

pmclanahan commented:
not sure why it's working locally actually

pmclanahan commented on bedrock/settings/base.py:
```diff
@@ -1138,6 +1146,16 @@ def JINJA_CONFIG():
+            'js/mozorg/content-services-form.js',
+            'js/base/mozilla-input-placeholder.js',
+        ),
+        'output_filename': 'js/partnerships-bundle.js',
```
This should get a unique bundle file name.

pmclanahan commented on bedrock/mozorg/views.py:
```diff
@@ -304,6 +305,88 @@ def partnerships(request):
+
+            honeypot = data.pop('office_fax')
+
+            if honeypot:
```
Wouldn't returning an error for the spam trap just tell the spammers that there are spam traps they need to fix? I think it'd be better if we just made it seem like it succeeded and not submit anything to SF.

sgarrity commented:
Ready for review!

jgmize commented:
@sgarrity please rebase, thanks

sgarrity commented:
> @sgarrity please rebase, thanks

Oops, the rebased version was in the mozilla/bedrock repo. I've pushed it here to my repo.

jpetto commented on bedrock/mozorg/forms.py:
```diff
@@ -506,6 +510,238 @@ def __init__(self, *args, **kwargs):
+        },
+        widget=forms.TextInput(
+            attrs={
+                'size': 20,
```
Should add `required` attributes as with other fields?

jpetto commented on bedrock/mozorg/templates/mozorg/contentservices/faq.html:
```diff
@@ -0,0 +1,247 @@
+    <div data-accordion-role="tabpanel">
+      <p>
+        {{ _('A Tile is a visual bookmark that consists of one or two images: a required “static state” image and an optional (but recommended) “rollover state.”') }}
+        {% trans url='#TODO' %}
```
Just calling out this `TODO`.

jpetto commented on bedrock/mozorg/templates/mozorg/contentservices/faq.html:
```diff
@@ -0,0 +1,247 @@
+    <h3 data-accordion-role="tab">{{ _('What are the technical requirements?') }}</h3>
+    <div data-accordion-role="tabpanel">
+      <p>
+      {% trans url='#TODO' %}
```
Another `TODO`.

jpetto commented on bedrock/mozorg/templates/mozorg/contentservices/about.html:
```diff
@@ -0,0 +1,53 @@
+{% block article %}
+  <h1 class="title-shadow-box">{{ self.page_title() }}</h1>
+
+
```
Nit - extra line break.

jpetto commented on bedrock/mozorg/templates/mozorg/contentservices/index.html:
```diff
@@ -0,0 +1,70 @@
+{% block site_css %}{% endblock %}
+
+{% block extrahead %}
+  {% stylesheet 'contentservices' %}
```
Is there a reason you're using the `extrahead` block here, but `page_css` in other files (like `start.html`)?

jpetto commented on bedrock/mozorg/templates/mozorg/contentservices/tiles.html:
```diff
@@ -0,0 +1,79 @@
+  </p>
+
+  <p>
+  {{ high_res_img('img/contentservices/suggested-tiles.jpg', {'alt': 'Suggested Tiles screenshot', 'width': '620'}) }}
```
Nit - indentation is inconsistent in some of these `<p>` tags.

jpetto commented on media/css/mozorg/contentservices.less:
```diff
@@ -0,0 +1,207 @@
+    float: right;
+    margin-top: -50px;
+    margin-bottom: @baseLine;
+    #welcome-video video {
```
Nit - don't really need to nest an id selector (though it is nice for organizational purposes).

jpetto commented on media/css/mozorg/contentservices.less:
```diff
@@ -0,0 +1,207 @@
+    h4 {
+        margin-bottom: @baseLine / 6;
+        a:after {
+            content: " »";
```
Use `trailing-arrow()` mixin?

jpetto commented on media/js/mozorg/content-services-form.js:
```diff
@@ -0,0 +1,122 @@
+
+    var campaign_type_other = function() {
+        return $campaign_type.val().indexOf('Other') > -1;
+        // TODO: Is this an l10n issue?
```
Just pointing out this `TODO`.

sgarrity replied:
This isn't a blocker for launch, but should be addressed before the page is localized.

jpetto commented on media/js/mozorg/content-services-form.js:
```diff
@@ -0,0 +1,122 @@
+    var $stateFlydown = $('#state');
+    var $provinceField = $('.province-field');
+    var $stateInput = $stateField.find('input, textarea, select');
+    var $provinceInput = $provinceField.find('input, textarea, select');
```
Doesn't look like `$stateInput` or `$provinceInput` are used anywhere?

jpetto commented on media/js/mozorg/content-services-form.js:
```diff
@@ -0,0 +1,122 @@
+
+    var country_is_us = function() {
+        return $country.val().indexOf('us') > -1;
+    }
```
Missing semicolon above.

jpetto commented on media/js/mozorg/content-services-form.js:
```diff
@@ -0,0 +1,122 @@
+                type: $sfForm.attr('method'),
+                dataType: 'json',
+                success: function(data, status, xhr) {
+                    $('#sf-form').fadeOut('fast', function() {
```
Could use the `$sfFrom` variable declare at the top?

jpetto commented on bedrock/mozorg/forms.py:
```diff
@@ -506,6 +510,238 @@ def __init__(self, *args, **kwargs):
+    def clean(self):
+        data = super(ContentServicesForm, self).clean()
+        if data.get('country') == 'us' and not data.get('state'):
+            raise ValidationError(self.fields['state'].error_messages['invalid'])
```
If JS is disabled (I know, I know) and the user is in the US, there is no way to avoid this error.

sgarrity replied:
I've updated the field to only be hidden if JS is enabled.

sgarrity commented:
@jpetto Thanks for the thorough review. I think I have everything addressed.

jpetto commented:
Things are looking good. Just needs a squash. r+ :octopus:

sgarrity commented:
Thanks @jpetto 

@pmclanahan How do you recommend I handle a squash since we both have commits in there?

sgarrity commented:
Squashed!

All of my commits and squashed into one commit, and all of @pmclanahan's into another commit.

@pmclanahan: I took some liberties combining your commit messages into a single commit message. Let me know if there are any issues there.
